spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -271,7 +271,7 @@ private SparkTableUtil() {
    * @param metricsConfig a metrics conf
    * @return a List of DataFile
    */
-  public static List<DataFile> listPartition(Map<String, String> partition, String uri, String format,
+  public static List<DataFile> listPartition(Map<String, String> partition, URI uri, String format,
```

Comment:
I'd like to avoid changing this method since it is public and using a URI will probably change behavior for users passing strings (String -> URI -> Path instead of String -> Path).